recipes/default.rb

```diff
@@ -10,102 +10,94 @@
 #
 
 # Global variable $gecos_os created to reduce calls to external programs
-$gecos_os = `lsb_release -d`.split(":")[1].chomp().lstrip()
-
-case node[:kernel][:machine]
-  when "x86_64"
-    $arch = "amd64"
-  when "i686"
-    $arch = "i386"
-  else  
-    $arch = ""
-end
-
-
-$gem_path = "/opt/chef/embedded/bin/gem"
-if not ::File.exist?($gem_path)
-    $gem_path = "/usr/bin/gem"
-end
-
+$gecos_os = `lsb_release -d`.split(':')[1].chomp.lstrip
+
+$arch = case node[:kernel][:machine]
+        when 'x86_64'
+          'amd64'
+        when 'i686'
+          'i386'
+        else
+          ''
+        end
+
+$gem_path = if ::File.exist?('/opt/chef/embedded/bin/gem')
+              '/opt/chef/embedded/bin/gem'
+            else
+              $gem_path = '/usr/bin/gem'
+            end
 
 # Snitch, the chef notifier has been renamed
 # TODO: move this to chef-client-wrapper
-if ::File.exists?("/usr/bin/gecos-snitch-client")
-  $snitch_binary="/usr/bin/gecos-snitch-client"
-else
-  $snitch_binary="/usr/bin/gecosws-chef-snitch-client"
-end  
+$snitch_binary = if ::File.exist?('/usr/bin/gecos-snitch-client')
+                   '/usr/bin/gecos-snitch-client'
+                 else
+                   '/usr/bin/gecosws-chef-snitch-client'
+                 end
 
-execute "gecos-snitch-client" do
+execute 'gecos-snitch-client' do
   command "#{$snitch_binary} --set-active true"
   action :nothing
 end.run_action(:run)
 
 # Prepare the environment variables
-Chef::Log.info("Prepare the environment variables for #{node['ohai_gecos']['pclabel']} computer")
-$gecos_environ = ENV.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}
+Chef::Log.info('Prepare the environment variables for '\
+    "#{node['ohai_gecos']['pclabel']} computer")
+$gecos_environ = ENV.each_with_object({}) { |(k, v), memo| memo[k.to_sym] = v }
 $gecos_environ['STATION'.to_sym] = node['ohai_gecos']['pclabel']
 $node = node
 
 # Get locale
 $locale = ENV['LANG'].downcase
-if $locale.include? "."
-  $locale = $locale.split('.')[0]
-end
+$locale = $locale.split('.')[0] if $locale.include? '.'
 Chef::Log.info("Locale is #{$locale}")
 
-
-# This should not be necessary, as wrapper is in new GECOS-Agent package. It is a transitional solution.
-Chef::Log.info("Installing wrapper")
-cookbook_file "gecos-chef-client-wrapper" do
-  path "/usr/bin/gecos-chef-client-wrapper"
+# This should not be necessary, as wrapper is in new GECOS-Agent package.
```

Comment:
It is a good moment to delete this duplicate wrapper creation.

Reply:
Ok. Removed